agent.go

```diff
@@ -31,12 +31,14 @@ type Agent struct {
 	lock  sync.Mutex
 	state agentState
 
+	cfgWatcher  ConfigWatcher
```

Comment:
Why do we have cfgWatcher, httpCfgWatcher, and memdCfgWatcher?

Reply:
The concrete classes have different configuration and reconfiguration requirements, but the interface is useful for most cases.

Reply:
Now that you mention it, I wonder if instead of exposing something like "WatchConfig()", if we should just expose the interfaces for all our internal components.